src/components/AddExpensesForm.js

```diff
@@ -0,0 +1,91 @@
+import { useState } from 'react';
+import { v4 as uuid } from 'uuid';
+
+const AddExpensesForm = () => {
+  const initialState = {
+    id: '',
+    name: '',
+    date: '',
+    amount: '',
+  };
+
+  const [singleExpense, setSingleExpense] = useState(initialState);
```

Comment:
why do we use a hook with an object over individual hooks?

Reply:
The `handleChange` function allow us to dynamically update our `singleExpense` state based on the key/value pairs coming from the inputs `name` and `value`. This is a shortcut that works as long as inputs and state can use the same `keys`. I personally like it to keep the code clean and well organised without abstracting too much.

Reply:
mmm, I personally prefer explicit code over implicit magic.

> I personally like it to keep the code clean and well organised without abstracting too much.
tip: comments like this can easily be misunderstood as an offense (because you are saying that the commenter is not interested in clean/well-organized code. i have seen people freaking out about these kind of comments. do you know what I mean?

Reply:
Nice! Happy!!